src/giphyService.ts

```diff
@@ -1,51 +1,52 @@
-import axios, {AxiosError, AxiosResponse} from 'axios';
+import axios, { AxiosError, AxiosResponse } from 'axios';
 
 export class GiphyService {
-
-    private static readonly giphyBaseUri: string = 'https://api.giphy.com/v1/gifs/search?api_key=';
-
-    private static readonly giphyLimit: number = 1;
-
-    private static readonly giphyRating: string = 'g';
-
-    private giphyApiKey: string = process.env.giphyApiKey;
-
-    constructor(giphyApiKey?: string) {
-        if (giphyApiKey) {
-            this.giphyApiKey = giphyApiKey;
-        }
+  private static readonly giphyBaseUri: string = 'https://api.giphy.com/v1/gifs/search?api_key=';
+  private static readonly giphyLimit: number = 9;
+  private static readonly giphyRating: string = 'g';
+  private giphyApiKey: string = process.env.giphyApiKey;
+
+  constructor(giphyApiKey?: string) {
+    if (giphyApiKey) {
+      this.giphyApiKey = giphyApiKey;
     }
-
-    public async getRandomGifUrl(query: string) {
-        const requestUri = this.createRequestUri(query);
-
-        let gifUrl: string;
-
-        try {
-            const result = await axios.get(requestUri).catch(this.handleError) as AxiosResponse;
-            console.log(result.data.data[0].images.original.url);
-            gifUrl = result.data.data[0].images.original.url;
-        } catch (e) {
-            this.handleError(e);
-        }
-
-        return gifUrl;
+  }
+
+  public async getRandomGifUrl(query: string) {
+    const requestUri = this.createRequestUri(query);
+
+    let gifUrl: string;
+    try {
+      const result = (await axios.get(requestUri).catch(this.handleError)) as AxiosResponse;
+      const gifs: any[] = result.data.data;
+
+      if (gifs.length > 0) {
```

Comment:
Can we move random number gen to another function? And then pass in giphyLimit?

Reply:
good idea, on it